### ISSUE TITLE
Crash with the Corpse

### ISSUE THREAD

**Issue by Redls07** [tr7zw/FirstPersonModel#233]
I play forge version 1.16.5 with optifine, and if I have a corpse mod, it crashes when I die.

**Comment by tr7zw** [comment]
Please also add the crash report to get an idea on what's happening

**Comment by Redls07** [comment]
---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 17.09.21 22:11
Description: Rendering entity in world

java.lang.NullPointerException: Rendering entity in world
	at net.minecraft.client.renderer.entity.PlayerRenderer.handler$zzm000$scale(PlayerRenderer.java:556) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225620_a_(PlayerRenderer.java) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225620_a_(PlayerRenderer.java:41) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.LivingRenderer.func_225623_a_(LivingRenderer.java:137) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:LivingEntityRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225623_a_(PlayerRenderer.java:64) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at de.maxhenkel.corpse.entities.CorpseRenderer.render(CorpseRenderer.java:64) ~[?:1.16.5-1.0.6] {re:classloading}
	at de.maxhenkel.corpse.entities.CorpseRenderer.func_225623_a_(CorpseRenderer.java:19) ~[?:1.16.5-1.0.6] {re:classloading}
	at net.minecraft.client.renderer.entity.EntityRendererManager.func_229084_a_(EntityRendererManager.java:311) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:EntityRendererMangerMixin,pl:mixin:A}
	at net.minecraft.client.renderer.WorldRenderer.func_228418_a_(WorldRenderer.java:2124) ~[?:?] {re:mixin,re:classloading,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:WorldRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.WorldRenderer.func_228426_a_(WorldRenderer.java:1740) ~[?:?] {re:mixin,re:classloading,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:WorldRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.GameRenderer.func_228378_a_(GameRenderer.java:1022) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.client.renderer.GameRenderer.func_195458_a(GameRenderer.java:693) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:976) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) [?:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) [forge-1.16.5-36.2.2.jar:36.2] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$456/1333945918.call(Unknown Source) [forge-1.16.5-36.2.2.jar:36.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.9.jar:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.client.renderer.entity.PlayerRenderer.handler$zzm000$scale(PlayerRenderer.java:556) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225620_a_(PlayerRenderer.java) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225620_a_(PlayerRenderer.java:41) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.entity.LivingRenderer.func_225623_a_(LivingRenderer.java:137) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:LivingEntityRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.entity.PlayerRenderer.func_225623_a_(PlayerRenderer.java:64) ~[?:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,xf:fml:forge:crossbows.2,pl:mixin:APP:firstpersonmod.forge.mixins.json:PlayerRendererMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at de.maxhenkel.corpse.entities.CorpseRenderer.render(CorpseRenderer.java:64) ~[?:1.16.5-1.0.6] {re:classloading}
	at de.maxhenkel.corpse.entities.CorpseRenderer.func_225623_a_(CorpseRenderer.java:19) ~[?:1.16.5-1.0.6] {re:classloading}
-- Entity being rendered --
Details:
	Entity Type: corpse:corpse (de.maxhenkel.corpse.entities.CorpseEntity)
	Entity ID: 1007
	Entity Name: Труп
	Entity's Exact location: 58.14, 68.00, -23.18
	Entity's Block location: World: (58,68,-24), Chunk: (at 10,4,8 in 3,-2; contains blocks 48,0,-32 to 63,255,-17), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null

-- Renderer details --
Details:
	Assigned renderer: de.maxhenkel.corpse.entities.CorpseRenderer@71d8794f
	Location: 0.00,-1.62,0.00 - World: (0,-2,0), Chunk: (at 0,-1,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Rotation: -122.34375
	Delta: 0.47999436
Stacktrace:
	at net.minecraft.client.renderer.entity.EntityRendererManager.func_229084_a_(EntityRendererManager.java:311) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:EntityRendererMangerMixin,pl:mixin:A}
	at net.minecraft.client.renderer.WorldRenderer.func_228418_a_(WorldRenderer.java:2124) ~[?:?] {re:mixin,re:classloading,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:WorldRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.WorldRenderer.func_228426_a_(WorldRenderer.java:1740) ~[?:?] {re:mixin,re:classloading,xf:OptiFine:default,pl:mixin:APP:firstpersonmod.forge.mixins.json:WorldRendererMixin,pl:mixin:A}
	at net.minecraft.client.renderer.GameRenderer.func_228378_a_(GameRenderer.java:1022) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}


-- Affected level --
Details:
	All players: 1 total; [ClientPlayerEntity['Jacque_Fresco_007'/329, l='ClientLevel', x=58.14, y=68.00, z=-23.18]]
	Chunk stats: Client Chunk Cache: 441, 289
	Level dimension: minecraft:overworld
	Level spawn location: World: (32,69,-32), Chunk: (at 0,4,0 in 2,-2; contains blocks 32,0,-32 to 47,255,-17), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Level time: 1948 game time, 1948 day time
	Server brand: forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.world.ClientWorld.func_72914_a(ClientWorld.java:617) ~[?:?] {re:classloading,xf:OptiFine:default}
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2029) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:623) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) [?:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) [forge-1.16.5-36.2.2.jar:36.2] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$456/1333945918.call(Unknown Source) [forge-1.16.5-36.2.2.jar:36.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.9.jar:?] {}


-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 14140171320 bytes (13485 MB) / 14931722240 bytes (14240 MB) up to 14931722240 bytes (14240 MB)
	CPUs: 12
	JVM Flags: 11 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -XX:+IgnoreUnrecognizedVMOptions -Xmn128M -Xmx14232M -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	ModLauncher: 8.0.9+86+master.3cf110c
	ModLauncher launch target: fmlclient
	ModLauncher naming: srg
	ModLauncher services: 
		/mixin-0.8.2.jar mixin PLUGINSERVICE 
		/eventbus-4.0.0.jar eventbus PLUGINSERVICE 
		/forge-1.16.5-36.2.2.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.16.5-36.2.2.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-3.0.1.jar accesstransformer PLUGINSERVICE 
		/forge-1.16.5-36.2.2.jar capability_inject_definalize PLUGINSERVICE 
		/forge-1.16.5-36.2.2.jar runtimedistcleaner PLUGINSERVICE 
		/mixin-0.8.2.jar mixin TRANSFORMATIONSERVICE 
		/OptiFine-OptiFine-1.16.5_HD_U_G8_pre2.jar OptiFine TRANSFORMATIONSERVICE 
		/forge-1.16.5-36.2.2.jar fml TRANSFORMATIONSERVICE 
	FML: 36.2
	Forge: net.minecraftforge:36.2.2
	FML Language Providers: 
		javafml@36.2
		minecraft@1
	Mod List: 
		forge-1.16.5-36.2.2-client.jar                    |Minecraft                     |minecraft                     |1.16.5              |DONE      |Manifest: NOSIGNATURE
		forge-1.16.5-36.2.2-universal.jar                 |Forge                         |forge                         |36.2.2              |DONE      |Manifest: 22:af:21:d8:19:82:7f:93:94:fe:2b:ac:b7:e4:41:57:68:39:87:b1:a7:5c:c6:44:f9:25:74:21:14:f5:0d:90
		notenoughanimations-1.2.4.jar                     |NotEnoughAnimations Mod       |notenoughanimations           |1.2.4               |DONE      |Manifest: NOSIGNATURE
		tl_skin_cape_forge_1.16.5-1.19.jar                |TLSkinCape                    |tlskincape                    |1.19                |DONE      |Manifest: 19:f5:ce:44:81:0c:e4:22:05:5e:73:c5:a8:cd:de:f3:c8:cf:a9:b3:01:70:40:a0:ee:2d:50:7a:1c:3d:1c:8a
		transliterationlib-1.0.4.jar                      |TRansliterationLib Mod        |transliterationlib            |1.0.4               |DONE      |Manifest: NOSIGNATURE
		firstpersonmod-2.0.1.jar                          |FirstPersonModel Mod          |firstpersonmod                |2.0.1               |DONE      |Manifest: NOSIGNATURE
		corpse-1.16.5-1.0.6.jar                           |Corpse                        |corpse                        |1.16.5-1.0.6        |DONE      |Manifest: NOSIGNATURE
	Crash Report UUID: 7bd2cc08-0701-4102-8ef3-330712b850eb
	Launched Version: ForgeOptiFine 1.16.5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Intel(R) UHD Graphics 630 GL version 4.6.0 - Build 27.20.100.9466, Intel
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'forge'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: vanilla, mod_resources
	Current Language: Русский (Россия)
	CPU: 12x Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz
	OptiFine Version: OptiFine_1.16.5_HD_U_G8_pre2
	OptiFine Build: 20210305-002749
	Render Distance Chunks: 8
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 - Build 27.20.100.9466
	OpenGlRenderer: Intel(R) UHD Graphics 630
	OpenGlVendor: Intel
	CpuCount: 12

**Comment by tr7zw** [comment]
That should be fairly easy to fix. The issue is that a player is rendered, while there is no player (cause he's dead).

**Comment by Redls07** [comment]
I will be very grateful if you fix it.

**Comment by tr7zw** [comment]
Check with the latest build https://github.com/tr7zw/FirstPersonModel/actions/runs/1255207386 (Download is the Artifacts button at the bottom)

**Comment by Redls07** [comment]
Unfortunately, when I start the game with this build, the game crashes on the loading screen. At the same time, no new files appear in the crash report folder.

**Comment by tr7zw** [comment]
It should be Forge and Fabric. Can you upload the latest.log after starting with that build?

**Comment by Redls07** [comment]
[latest.log](https://github.com/tr7zw/FirstPersonModel/files/7202233/latest.log)

**Comment by tr7zw** [comment]
Thaaats... Odd? Somehow a method is missing in the WorldRenderer. My only guess is that Optifine changes something or that something else in the build broke. Need to check when I'm home to compare/test myself.

**Comment by Redls07** [comment]
I just tried to run the forge without optifine, but the crash is still present. If necessary, here is the startup log without optifi
[latest.log](https://github.com/tr7zw/FirstPersonModel/files/7202420/latest.log)
ne

**Comment by tr7zw** [comment]
There it wasn't finding something else. Damn Forge 1.16 is a pain in the ass. I think I'll redo the 1.17 branch with mojmap and loom, and then backport it to 1.16.

**Comment by Redls07** [comment]
Ok, thanks. Oh, and I just found out that if the launch occurs without optifine, then the files in the crash log folder appear, so if this is important, then here
[crash-2021-09-21_10.38.36-client.txt](https://github.com/tr7zw/FirstPersonModel/files/7202438/crash-2021-09-21_10.38.36-client.txt)

**Comment by Redls07** [comment]
But I just found out that they are essentially the same thing, so never mind lol

**Comment by tr7zw** [comment]
Ok can confirm that it does not work. The issue is I can't do local test builds because of the entire thing not running under Java 16. So I'll do the plan with making 1.17 with mojmap and forge loom, and then backport that to 1.16(maybe even 1.15?)

**Comment by tr7zw** [comment]
In other news: the current build on curse for 1.16 seems to be absolutely fine with the corpse mod? Even when using the scaling function that had the issue in the original report 
![grafik](https://user-images.githubusercontent.com/5133936/134399643-ed4dc6d3-1044-479a-9068-465f506cef98.png)

**Comment by Redls07** [comment]
Hm, interesting. I tried to re-download the latest versions of corpse and fpm from curseforge, just in case, even though I already had the latest versions of these mods. But no, the crash is still repeated. And I also tried it without optifine, and the crash is still present. I do not know what could be the reason for this.

**Comment by tr7zw** [comment]
Maybe an old Forge install or other mods. But with 1.16.5 Forge-36.2.0 and the cropse mod 1.0.5, First person mod 2.0.1, Not enough animations, skin layers 3d and transliteration lib it seems to work from what I can tell.

**Comment by Redls07** [comment]
I still managed to fix the crash. I removed from my mods folder all the mods that were there, with the exception of those that I am testing. But there was also a mod that didn't get deleted. It was a tl_skin_cape that was built into the TLauncher that I was using. I thought that the problem might be in it, so I deleted this launcher, and transferred everything I needed to another launcher, where the tl_skin_cape was not built in. And yes, it worked. The crash is gone. Thank you for helping me figure out all this.
